tests/jlm/util/TestHashSet.cpp

```diff
@@ -180,3 +180,41 @@ TestIntersectWith()
 }
 
 JLM_UNIT_TEST_REGISTER("jlm/util/TestHashSet-TestIntersectWith", TestIntersectWith)
+
+static int
+TestDifferenceWith()
+{
+  using namespace jlm::util;
+
+  HashSet<int> set12({ 1, 2 });
+  HashSet<int> set123({ 1, 2, 3 });
+  HashSet<int> set45({ 4, 5 });
+
+  set123.DifferenceWith(set12); // {1, 2, 3} - {1, 2}
+  assert(set123.Size() == 1);
+  assert(set123.Contains(3));
+
+  // set12 was not touched
+  assert(set12.Size() == 2);
```

Comment:
The  better way to check this is to make a copy of set12 and then equal the copy to the set12 after the operation.